Feature: Signal Conditioning - Current Limiter

GabrieleGandolfi opened the pull request:
This PR introduces a current limiter designed to protect motors from operating beyond their rated conditions for prolonged periods of time.
The limiter allows intermittent operation beyond the nominal values, but as soon as these are exceeded, it gradually introduces corrective action aimed at smoothly returning the current values to their rated levels.
The user can set the threshold value based on the particular motor used, and can also set gain values to adjust how timely and aggressive the limiter's corrective action should be.

Here is an example of the effect of the current limiter in maintaining the motor current at the nominal value of 0.343 A for continuous use.

<img width="2392" height="1860" alt="image" src="https://github.com/user-attachments/assets/bda5f4a9-b790-472a-9ce2-14d4a6d0a62d" />

GabrieleGandolfi committed:
Added current limiter test

GabrieleGandolfi committed:
Added docs

GabrieleGandolfi committed:
Update README.md

GabrieleGandolfi committed:
Update README.md

GabrieleGandolfi committed:
Updates after first review. Issue #3

GabrieleGandolfi committed:
Added the units explicitly in the prompt of the block mask, as suggested in Issue #3

GabrieleGandolfi committed:
Renamed folder to Signal Conditioning.

GabrieleGandolfi committed:
Solved conflict in README.md

pattacini commented on docs/signal_conditioning/current_limiter/README.asv:
Please remove this file.

pattacini commented on tests/test_current_limiter.slx.r2025a:
Please remove this file.

pattacini reviewed (changes requested):
Hi @GabrieleGandolfi 

There must be a problem with this PR as I don't see any modification applied to the arsenal model.

Also, there are a couple of files to be removed at least.

GabrieleGandolfi committed:
Removed .asv and .r2025a files.

GabrieleGandolfi commented:
> Hi @GabrieleGandolfi
> 
> There must be a problem with this PR as I don't see any modification applied to the arsenal model.

True, I just added the block to the arsenal.slx model.

> Also, there are a couple of files to be removed at least.

I also deleted these two files.

GabrieleGandolfi committed:
Solved README.md conflict.

GabrieleGandolfi committed:
Solved library conflict.

GabrieleGandolfi committed:
Update README.

pattacini committed:
adjustments to current limiter

pattacini reviewed (approved):
Hi @GabrieleGandolfi 

I've made a few minor adjustments:
- Applied a description to the block.
- Introduced a view to launch SDI at the end of the test.

Nice work!
Merging the PR.
